common/lib/xmodule/xmodule/tests/test_capa_module.py

```diff
@@ -361,6 +361,65 @@ def test_showanswer_past_due(self):
+        the show answer button should
+        only appear after the user has
+        attempted the problem(i.e clicked the
+        submit button) for the requisite number of times.
+
+        Once the button appears, it does not disappear again.
+        """
+        # Cannot see answer because user has not attempted
+        # answering question for the required number of
+        # times.
+        before_attempts_with_max = CapaFactory.create(showanswer="after_attempts",
+                                                      attempts="2",
+                                                      attempts_before_showanswer_button="3",
+                                                      max_attempts="5")
+        self.assertFalse(before_attempts_with_max.answer_available())
+        # can't see answer even when max_attempts is not set.
+        before_attempts_no_max = CapaFactory.create(showanswer="after_attempts",
+                                                    attempts="2",
+                                                    attempts_before_showanswer_button="3")
+        self.assertFalse(before_attempts_no_max.answer_available())
+        # can see answer even after all attempts are used up,
+        # even with the due date in the future
+        used_all_attempts = CapaFactory.create(showanswer='after_attempts',
+                                               attempts_before_showanswer_button="2",
+                                               max_attempts="3",
+                                               attempts="3",
+                                               due=self.tomorrow_str)
+        self.assertTrue(used_all_attempts.answer_available())
+        # can see after due date
+        past_due_date = CapaFactory.create(showanswer='after_attempts',
+                                           attempts_before_showanswer_button="2",
+                                           attempts="2",
+                                           due=self.yesterday_str)
+        self.assertTrue(past_due_date.answer_available())
+        # Can see even though grace period hasn't expired as long as
+        # User has already attempted for the required number of times
+        still_in_grace = CapaFactory.create(showanswer='after_attempts',
+                                            after_attempts="3",
+                                            attempts="4",
+                                            due=self.yesterday_str,
+                                            graceperiod=self.two_day_delta_str)
+        self.assertTrue(still_in_grace.answer_available())
+        # Ensure that if attempts_before_showanswer_button > max_attempts, the button shows up
+        # after all attempts are used up, i.e after_attempts becomes max_attempts
+        large_after_attempts = CapaFactory.create(showanswer='after_attempts',
+                                                  attempts_before_showanswer_button="5",
```

Comment:
Here and throughout, we're switching between single and double quotes.

For consistency, let's just use single quotes.

Note: this comment only applies to the new code you've added here;
don't change the existing code.